cupy/_core/dlpack.pyx

```diff
@@ -19,139 +14,213 @@ import warnings
 import cupy
 import cupy._core.core as core
 
+from .dlpack cimport *
 
-cdef extern from './include/cupy/_dlpack/dlpack.h' nogil:
-    cdef int DLPACK_VERSION
-
-    cdef enum DLDeviceType:
-        kDLCPU
-        kDLCUDA
-        kDLCUDAHost
-        kDLOpenCL
-        kDLVulkan
-        kDLMetal
-        kDLVPI
-        kDLROCM
-        kDLROCMHost
-        kDLExtDev
-        kDLCUDAManaged
-        kDLOneAPI
-        kDLWebGPU
-        kDLHexagon
-
-    ctypedef struct DLDevice:
-        DLDeviceType device_type
-        int32_t device_id
-
-    cdef enum DLDataTypeCode:
-        kDLInt
-        kDLUInt
-        kDLFloat
-        kDLBfloat
-        kDLComplex
-        kDLBool
-
-    ctypedef struct DLDataType:
-        uint8_t code
-        uint8_t bits
-        uint16_t lanes
-
-    ctypedef struct DLTensor:
-        void* data
-        DLDevice device
-        int32_t ndim
-        DLDataType dtype
-        int64_t* shape
-        int64_t* strides
-        uint64_t byte_offset
-
-    ctypedef struct DLManagedTensor:
-        DLTensor dl_tensor
-        void* manager_ctx
-        void (*deleter)(DLManagedTensor*)  # noqa: E211
-
-
-def get_build_version():
-    return str(DLPACK_VERSION)
+
+cdef const char* CAPSULE_NAME = "dltensor"
+cdef const char* CAPSULE_NAME_VER = "dltensor_versioned"
+cdef const char* USED_CAPSULE_NAME = "used_dltensor"
+cdef const char* USED_CAPSULE_NAME_VER = "used_dltensor_versioned"
```

Comment:
Q: I was under the impression that Cython 0.29.x does not support `cdef const`? Does this compile fine? (CuPy has not migrated yet: #4610)

Reply:
Yeah, compiled fine.  Was it just that it ignored it, so in C it is gone (I recall issues, but honestly not what)?